README.md

```diff
@@ -1,78 +1,406 @@
+      ]
+    },
+    {
+      "constraints": [
+        { "field": "age", "is": "ofType", "value": "integer" },
+        { "field": "age", "is": "greaterThan", "value": 0 },
+        { "field": "age", "is": "lessThan", "value": 100 },
+        { "not": { "field": "age", "is": "null" } }
+      ]
+    }
+  ]
+}
+~~~
+
+With the change in the `firstname` regex, the output is now starting to look quite realistic:
+
+~~~
+firstName,age
+"John",31
+"Mary",73
+"Jon",44
+"John",24
+"Jon",43
+"Jon",18
+"Jon",5
+"John",82
+"John",19
+"Jon",43
+[...]
+~~~
+
+Finally, before exploring some more interesting features of the generator, we'll add a field for National Insurance number. In this case, the constraints applied to the field ensure that it only has a value if the age is greater than 15:
+
+~~~json
+{
+  "schemaVersion": "0.1",
+  "fields": [
+    { "name": "firstName" },
+    { "name": "age" },
+    { "name": "nationalInsurance" }
+  ],
+  "rules": [
+    {
+      "rule": "first name",
+      "constraints": [
+        { "field": "firstName", "is": "ofType", "value": "string" },
+        { "not": { "field": "firstName", "is": "null" } },
+        { "field": "firstName", "is": "matchingRegex", "value": "(Joh?n|Mar[yk])" }
+      ]
+    },
+    {
+      "rule": "age",
+      "constraints": [
+        { "field": "age", "is": "ofType", "value": "numeric" },
+        { "field": "age", "is": "greaterThan", "value": 0 },
+        { "field": "age", "is": "lessThan", "value": 100 },
+        { "not": { "field": "age", "is": "null" } }
+      ]
+    },
+    {
+      "rule": "national insurance",
+      "constraints": [
+        { "field": "nationalInsurance", "is": "ofType", "value": "string" },
+        { "not": { "field": "nationalInsurance", "is": "null" } },
+        {
+          "if": {
+            "field": "age",
+            "is": "greaterThan",
+            "value": 15
+          },
+          "then": {
+            "field": "nationalInsurance",
+            "is": "matchingRegex",
+            "value": "[A-CEGHJ-PR-TW-Z]{1}[A-CEGHJ-NPR-TW-Z]{1}[0-9]{6}[A-DFM]{0,1}"
+          },
+          "else": {
+            "field": "nationalInsurance",
+            "is": "equalTo",
+            "value": "N/A"
+          }
+        }
+      ]
+    }
+  ]
+}
+~~~
+
+The above profile yields the following output:
+
+~~~
+firstName,age,nationalInsurance
+"Mary",85,"RP139559B"
+"Jon",2,"N/A"
+"Mark",10,"N/A"
+"Mary",20,"NW372471"
+"Mark",47,"ZP195766"
+"Jon",41,"RW557276F"
+"Mark",16,"N/A"
+"Jon",69,"OR650886"
+"John",19,"CB460473"
+"Mark",55,"AE656210B"
+"Mark",68,"ER027015"
+[...]
+~~~
+
+You can find out more about the various constraints the generator supports in the detailed [Profile Developer Guide](???).
+
+## Generation modes
+
+The generator supports a number of different generation modes:
+
+ - **random** - generates random data that abides by the given set of constraints, with the potentially infinite output limited via the `--num-rows` option.
```

Comment:
The switch is currently `--max-rows`.

Also, if this switch isn't supplied you'd get 1000 rows of random data, there is no (longer the) possibility of generating infinite amounts of random data